python/activator/middleware_interface.py

```diff
@@ -125,15 +125,18 @@ def __init__(self, central_butler: Butler, image_bucket: str, instrument: str,
         self._init_ingester()
         self._init_visit_definer()
 
-        # TODO DM-34098: note that we currently need to supply instrument here.
         # HACK: explicit collection gets around the fact that we don't have any
         # timestamp/exposure information in a form we can pass to the Butler.
         # This code will break once cameras start being versioned.
         self.camera = self.central_butler.get(
             "camera", instrument=self.instrument.getName(),
             collections=self.instrument.makeUnboundedCalibrationRunName()
         )
-        self.skymap = self.central_butler.get("skyMap")
+        # TODO: is central_butler guaranteed to have only one skymap dimension?
```

Comment:
Given the assert below, I think you don't need this TODO. We're defining the central repo as having exactly one skymap as part of this, and I think that's something that we as the AP group can ensure.

Reply:
I don't see why the `assert` makes the TODO irrelevant. The `assert` is computer-readable but easy for a human to overlook.

I'm not so confident that we will only have one skymap -- I can see a lot of experimentation over which skymap gives e.g. the best performance, especially during commissioning, and there might even be a policy some day that different datasets or data sources are based on different skymaps. Ultimately, it's not our call, but K-T's and the commissioning team's.